tsdb/index/tsi1/partition_test.go

```diff
@@ -82,6 +83,73 @@ func TestPartition_Manifest(t *testing.T) {
 	})
 }
 
+var badManifestPath string = filepath.Join(os.DevNull, tsi1.ManifestFileName)
+
+func TestPartition_Manifest_Write_Fail(t *testing.T) {
+	const expectedError = "not a directory"
+	t.Run("write MANIFEST", func(t *testing.T) {
+		m := tsi1.NewManifest(badManifestPath)
+		_, err := m.Write()
+		if !strings.Contains(err.Error(), expectedError) {
+			t.Fatalf("expected: %q, got: %q", expectedError, err.Error())
+		}
+	})
+}
+
+func TestPartition_PrependLogFile_Write_Fail(t *testing.T) {
+	const expectedError = "not a directory"
+	t.Run("write MANIFEST", func(t *testing.T) {
+		sfile := MustOpenSeriesFile()
+		defer sfile.Close()
+
+		p := MustOpenPartition(sfile.SeriesFile)
+		defer func() {
+			if err := p.Close(); err != nil {
+				t.Fatalf("error closing partition: %v", err)
+			}
+		}()
+		p.Partition.MaxLogFileSize = -1
+		fileN := p.FileN()
+		p.CheckLogFile()
+		if fileN >= p.FileN() {
+			t.Fatalf("manifest write prepending log file should have succeeded but number of files did not change correctly: expected more than %d files, got %d files", fileN, p.FileN())
```

Comment:
```suggestion
			t.Fatalf("manifest write prepending log file should have succeeded but number of files did not change correctly: expected less than %d files, got %d files", fileN, p.FileN())
```

Reply:
This one is correct, I believe. We expected more than `FileN`, but `FileN` is larger than or equal to `p.FileN()`

Reply:
You're right!